model/jpa/src/main/java/org/keycloak/organization/jpa/JpaOrganizationProvider.java

```diff
@@ -249,15 +255,18 @@ public boolean isManagedMember(OrganizationModel organization, UserModel member)
             return false;
         }
 
-        IdentityProviderModel identityProvider = organization.getIdentityProvider();
+        List<IdentityProviderModel> brokers = organization.getIdentityProviders().toList();
 
-        if (identityProvider == null) {
+        if (brokers.isEmpty()) {
             return false;
         }
 
-        FederatedIdentityModel federatedIdentity = userProvider.getFederatedIdentity(realm, member, identityProvider.getAlias());
+        List<FederatedIdentityModel> federatedIdentities = userProvider.getFederatedIdentitiesStream(realm, member)
+                .map(federatedIdentityModel -> realm.getIdentityProviderByAlias(federatedIdentityModel.getIdentityProvider())).filter(brokers::contains)
+                .map(m -> userProvider.getFederatedIdentity(realm, member, m.getAlias()))
+                .toList();
 
-        return federatedIdentity != null;
+        return !federatedIdentities.isEmpty();
```

Comment:
Perhaps we could avoid collecting streams to collections here. Something like might do the trick,. wdyt?

```java
return userProvider.getFederatedIdentitiesStream(realm, member)
                .map(federatedIdentityModel -> realm.getIdentityProviderByAlias(federatedIdentityModel.getIdentityProvider()))
                .filter(idp -> organization.getIdentityProviders().anyMatch(broker -> broker.getAlias().equals(idp.getAlias())))
                .map(idp -> userProvider.getFederatedIdentity(realm, member, idp.getAlias()))
                .anyMatch(Objects::nonNull);
```

Reply:
I would keep it as is. No need to resolve federated identities if there is no brokers associated with the orgs.